docs/DeveloperGuide.md

```diff
@@ -163,6 +163,60 @@ In order to mark a task as completed, the user keys in a valid command (e.g. `ma
+
+The sequence diagram is similar apart from:
+1. the command executed and parsed (`addmc l/mastery check 1 sd/2022-09-15T20:00 ed/2022-09-15T20:30 si/1 si/2` instead of `deletestudent 2`)
+2. the different command class (`AddMasteryCheckCommandParser` and `AddMasteryCheckCommand` instead of `DeleteStudentCommandParser` and `DeleteStudentCommand`)
+3. function called in main (`addLesson()` instead of `deleteStudent`)
+
+More detailed implementation of the parsing of user input is be shown in the activity diagram below.
+
+<img src="images/ParseMasteryCheckActivityDiagram.png" width="550"/>
+
+From the diagram, `MasteryCheckCommandParser` checks if
+
+1. all prefix are present
+2. lesson description is empty
+3. start date time is before end date time
+4. student index are int
+
+The rationale behind this design is that for all `Lesson`, there must be a `LessonDesc` present.
+It is also illogical for a lesson to start after the end date time. A `Student` must also be assigned manually to a `MasteryCheck`
+as the purpose of `MasteryCheck` is to assess a student's capability.
+
+**Future Implementation**
+- Allow user to input duration of lesson(in hours) to replace end date time
+- JARVIS will calculate the end date time for user based on the given start date time and duration
+- Helps to shorten the command required to be typed
+
+
+The following sequence diagram shows what happens when the AddMasteryCheckCommand is executed upon a successful command.
+
+<img src="images/AddMasteryCheckSequenceDiagram.png" width="550"/>
+
+What is not shown in the above sequence diagram are:
+
+1. how `AddMasteryCheckCommand` creates a `MasteryCheck` object with students when `AddMasteryCheckCommandParser` returns parses student indexes
+2. considerations when a `MasteryCheck` object is created but not added to `Model`
+
+`AddMasteryCheckCommand` will get the students involved in the `MasteryCheck` via indexing of the `lastShownList`.
+
+After a `MasteryCheck` object is created, `Model` will check if there already exists a `MasteryCheck` in the current `LessonBook` with the same identity fields.
+`Model` will also check with existing `Lessons` if there will be a clash in `TimePeriod`. This serves as a reminder to the user that there is already another lesson at that time.
+
+The above explanation is also applicable to adding consultation and studio lessons.
+They are similar apart from:
+1. the different naming(`AddConsultCommandParser`, `AddStudioCommandParser`, etc instead of `AddMasteryCheckParser`)
+2. for `Studio`, all `Student` currently in the `StudentBook` instead of `FilteredStudentList` will be used to create `LessonAttendance` and `LessonNotes`
+   1. Studio are tutorials and all students are expected to attend. Should any student not attend, the user can simply mark the student as absent.
```

Comment:
The second line seems to imply that students are marked present by default for studios which is not the case.